### PR TITLE
Add UseDotNet for other packaging pipelines

### PR DESCRIPTION
**Pull request by YUNQIUGUO**
### Motivation and Context

Fix Zip-Nuget-packaging pipeline: (following the updates of .NET 6.0->7.0)

https://aiinfra.visualstudio.com/Lotus/_build/results?buildId=448708&view=logs&j=434e4c35-3530-54f8-92f5-111e7141d27c&t=9503154a-1a4d-589b-7d54-4bceac8a03e1&l=30

### CODE REVIEW

**Comment by YUNQIUGUO** [comment]
There are two failures here:

The current change in this pr should fix this:

https://aiinfra.visualstudio.com/Lotus/_build/results?buildId=448708&view=logs&j=434e4c35-3530-54f8-92f5-111e7141d27c&t=9503154a-1a4d-589b-7d54-4bceac8a03e1&l=30

##[error]C:\Program Files (x86)\dotnet\sdk\6.0.421\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.TargetFrameworkInference.targets(144,5): Error NETSDK1045: The current .NET SDK does not support targeting .NET 7.0.  Either target .NET 6.0 or lower, or use a version of the .NET SDK that supports .NET 7.0.

however, I am not sure about how to fix the other one happened in e2e C# test: (involving both the Zip-Nuget packaging pipeline and the Nuget-CUDA packaging pipeline):

https://aiinfra.visualstudio.com/Lotus/_build/results?buildId=448708&view=logs&j=12994cf6-ee01-5d5d-5fbb-c4c9ef2986b7&t=3e05170d-9d6a-56fc-96fb-7e91290e1fc0&l=1246

Error message looks like the E2E test app is not compatible with .NET framework after upgrading the TFM from net6.0->net7.0 in .csproj:

_2024-04-22T13:41:58.5476362Z   Microsoft.ML.OnnxRuntime.EndToEndTests -> D:\a\_work\1\s\csharp\test\Microsoft.ML.OnnxRuntime.EndToEndTests\bin\Debug\net7.0\win-x64\Microsoft.ML.OnnxRuntime.EndToEndTests.dll
2024-04-22T13:41:58.5610519Z Test run for D:\a\_work\1\s\csharp\test\Microsoft.ML.OnnxRuntime.EndToEndTests\bin\Debug\net7.0\win-x64\Microsoft.ML.OnnxRuntime.EndToEndTests.dll (.NETCoreApp,Version=v7.0)
2024-04-22T13:41:58.8112925Z Microsoft (R) Test Execution Command Line Tool Version 17.8.0 (x64)
2024-04-22T13:41:58.8113471Z Copyright (c) Microsoft Corporation.  All rights reserved.
2024-04-22T13:41:58.8134892Z 
2024-04-22T13:41:59.9841323Z Starting test execution, please wait...
2024-04-22T13:42:00.1376874Z A total of 1 test files matched the specified pattern.
2024-04-22T13:42:00.2226588Z Testhost process for source(s) 'D:\a\_work\1\s\csharp\test\Microsoft.ML.OnnxRuntime.EndToEndTests\bin\Debug\net7.0\win-x64\Microsoft.ML.OnnxRuntime.EndToEndTests.dll' exited with error: You must install or update .NET to run this application.
2024-04-22T13:42:00.2227626Z App: D:\a\_work\1\s\csharp\test\Microsoft.ML.OnnxRuntime.EndToEndTests\bin\Debug\net7.0\win-x64\testhost.dll
2024-04-22T13:42:00.2227902Z Architecture: x64
2024-04-22T13:42:00.2228145Z Framework: 'Microsoft.NETCore.App', version '7.0.0' (x64)
2024-04-22T13:42:00.2228321Z .NET location: C:\Program Files\dotnet\
2024-04-22T13:42:00.2228474Z The following frameworks were found:
2024-04-22T13:42:00.2228638Z   6.0.25 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2024-04-22T13:42:00.2228822Z   8.0.0 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2024-04-22T13:42:00.2228970Z Learn more:
2024-04-22T13:42:00.2229158Z https://aka.ms/dotnet/app-launch-failed
2024-04-22T13:42:00.2229318Z To install missing framework, download:
2024-04-22T13:42:00.2229585Z https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=7.0.0&arch=x64&rid=win-x64&os=win10
2024-04-22T13:42:00.2229790Z . Please check the diagnostic logs for more information.
2024-04-22T13:42:00.2427903Z 
2024-04-22T13:42:00.2449698Z Test Run Aborted._

@skottmckay @snnn any ideas maybe? appreciate a lot in advance

**Comment by snnn** [comment]
I am trying to reproduce the problem.

File: tools/ci_build/github/azure-pipelines/nuget/templates/dml-vs-2022.yml [hunk 1]
```diff
@@ -76,6 +76,13 @@ stages:
         inputs:
           versionSpec: '18.x'
 
+      - task: UseDotNet@2
+        inputs:
+          version: '7.x'
+        ${{ if eq(parameters.BuildArch, 'x86') }}:
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/nuget/templates/dml-vs-2022.yml:82]: .Net 7 is EOL. We should use .Net 8

**Pull request closed without merging**